app/HutchManager/Controllers/AgentsController.cs

```diff
@@ -17,8 +17,9 @@ public AgentsController(DataSourceService dataSources)
   }
 
   [HttpPost]
-  public async Task<List<DataSource>> CheckIn(List<DataSource> dataSource)
+  public async Task<List<DataSource>> CheckIn(List<string> stringList)
```

Comment:
the interface of the api is not quite right here.

it wants to receive an `AgentCheckInModel` really, which will have one property currently (but maybe others in future):

```csharp
public class AgentCheckInModel
{
  public List<string> DataSources { get; set; } = new();
}
```

that will line up with the payload the agent sends:

```jsonc
{
  "dataSources": [
    "string",
    // ...
  ]
}
```